### ISSUE TITLE
Add `Sensitivity` metric.

### ISSUE THREAD

**Issue by stancld** [PyTorchLightning/metrics#976]
## 🚀 Feature

Add `Sensitivity` metric, which is a close one to [`Specificity`](https://torchmetrics.readthedocs.io/en/stable/classification/specificity.html#specificity) metric already implemented in `Torchmetrics`.

### Motivation

To cover another classification metric. Also, this allows us to implement `SpecificityAtSensitivity` and `SensitivityAtSpecificity` proposed in #965.

### Pitch

Add `Sensitivity` (implementation almost fully follows the `Specificity` except for the very last calculation).

### Additional context

Good first issue.

**Comment by Akshay1-6180** [comment]
Not sure if its integrated or now , if anyone is looking for a simple fix just call from this code 


```
import torch
from torch import Tensor
from typing_extensions import Literal

from torchmetrics.classification.stat_scores import BinaryStatScores
from torchmetrics.metric import Metric
from typing import Optional
from torchmetrics.utilities.compute import _safe_divide


class BinarySensitivity(BinaryStatScores):
    r"""Computes `Specificity`_ for binary tasks:

    .. math:: \text{Specificity} = \frac{\text{TN}}{\text{TN} + \text{FP}}

    Where :math:`\text{TN}` and :math:`\text{FP}` represent the number of true negatives and
    false positives respecitively.

    As input to ``forward`` and ``update`` the metric accepts the following input:

    - ``preds`` (:class:`~torch.Tensor`): An int or float tensor of shape ``(N, ...)``. If preds is a floating point
      tensor with values outside [0,1] range we consider the input to be logits and will auto apply sigmoid per
      element. Addtionally, we convert to int tensor with thresholding using the value in ``threshold``.
    - ``target`` (:class:`~torch.Tensor`): An int tensor of shape ``(N, ...)``

    As output to ``forward`` and ``compute`` the metric returns the following output:

    - ``bs`` (:class:`~torch.Tensor`): If ``multidim_average`` is set to ``global``, the metric returns a scalar value.
      If ``multidim_average`` is set to ``samplewise``, the metric returns ``(N,)`` vector consisting of a scalar value
      per sample.

    Args:
        threshold: Threshold for transforming probability to binary {0,1} predictions
        multidim_average:
            Defines how additionally dimensions ``...`` should be handled. Should be one of the following:

            - ``global``: Additional dimensions are flatted along the batch dimension
            - ``samplewise``: Statistic will be calculated independently for each sample on the ``N`` axis.
              The statistics in this case are calculated over the additional dimensions.

        ignore_index:
            Specifies a target value that is ignored and does not contribute to the metric calculation
        validate_args: bool indicating if input arguments and tensors should be validated for correctness.
            Set to ``False`` for faster computations.

    Example (preds is int tensor):
        >>> from torchmetrics.classification import BinarySpecificity
        >>> target = torch.tensor([0, 1, 0, 1, 0, 1])
        >>> preds = torch.tensor([0, 0, 1, 1, 0, 1])
        >>> metric = BinarySpecificity()
        >>> metric(preds, target)
        tensor(0.6667)

    Example (preds is float tensor):
        >>> from torchmetrics.classification import BinarySpecificity
        >>> target = torch.tensor([0, 1, 0, 1, 0, 1])
        >>> preds = torch.tensor([0.11, 0.22, 0.84, 0.73, 0.33, 0.92])
        >>> metric = BinarySpecificity()
        >>> metric(preds, target)
        tensor(0.6667)

    Example (multidim tensors):
        >>> from torchmetrics.classification import BinarySpecificity
        >>> target = torch.tensor([[[0, 1], [1, 0], [0, 1]], [[1, 1], [0, 0], [1, 0]]])
        >>> preds = torch.tensor(
        ...     [
        ...         [[0.59, 0.91], [0.91, 0.99], [0.63, 0.04]],
        ...         [[0.38, 0.04], [0.86, 0.780], [0.45, 0.37]],
        ...     ]
        ... )
        >>> metric = BinarySpecificity(multidim_average='samplewise')
        >>> metric(preds, target)
        tensor([0.0000, 0.3333])
    """

    def compute(self) -> Tensor:
        tp, fp, tn, fn = self._final_state()
        return _sensitivity_reduce(
            tp, fp, tn, fn, average="binary", multidim_average=self.multidim_average
        )


def _sensitivity_reduce(
    tp: Tensor,
    fp: Tensor,
    tn: Tensor,
    fn: Tensor,
    average: Optional[Literal["binary", "micro", "macro", "weighted", "none"]],
    multidim_average: Literal["global", "samplewise"] = "global",
) -> Tensor:
    if average == "binary":
        return _safe_divide(tp, tp + fn)
    elif average == "micro":
        tp = tp.sum(dim=0 if multidim_average == "global" else 1)
        fn = fn.sum(dim=0 if multidim_average == "global" else 1)
        return _safe_divide(tp, tp + fn)
    else:
        sensitivity_score = _safe_divide(tp, tp + fn)
        if average is None or average == "none":
            return sensitivity_score
        if average == "weighted":
            weights = tn + fp
        else:
            weights = torch.ones_like(sensitivity_score)
        return _safe_divide(
            weights * sensitivity_score, weights.sum(-1, keepdim=True)
        ).sum(-1)
```
